projects/plugins/jetpack/extensions/blocks/premium-content/edit.js

```diff
@@ -217,6 +218,44 @@ function Edit( props ) {
 
 	const { isSelected, className } = props;
 
+	useEffect( () => {
```

Comment:
This method only runs when re-rendering the premium-content block itself *not* it's children, in terms of
selection that is when selecting this block and when moving selection away from it.

This is still the case even when removing the dependencies below.

Reply:
This happens because `getSelectedBlock` is called from inside `useEffect`.

`useEffect` only runs when one of its dependencies changes (in this case: the block's own `clientId`, and `isSelected`).
To trigger the hook whenever the block selection changes, we need to call the selector from outside and pass its result as `useEffect` dependency.

E.g.
```es6
import { useSelect } from '@wordpress/data';

const selectedBlock = useSelect( select => select( 'core/block-editor' ).getSelectedBlock() );

useEffect( () => {
  if ( ! selectedBlock ) return;
  // Proceed with the whole block parents logic
}, [ selectedBlock ] );
```

Reply:
That was exactly it. Thank you.